feat: Add jmxtrans deployment (#2339)
strimzi/strimzi-kafka-operator: documentation/modules/appendix_crds.adoc

|string
|====

[id='type-JmxTransSpec-{context}']
### `JmxTransSpec` schema reference

Used in: xref:type-KafkaJmxOptions-{context}[`KafkaJmxOptions`]

[options="header"]
|====
|Property                  |Description

> |outputDefinitions  1.2+<.<|Defines the output hosts that will be referenced later on. For more information, see `proc-jmxtrans-deployment.adoc`.

Reviewer: The link won't work. We tend to reference the schema from the main part of the guide rather than the other way around. So maybe we can drop this link and refer back to here from the procedure? Maybe also we should say more about 'output hosts' in the procedure.

Reviewer: I mentioned above, but something like:

For more information on these properties see, xref:type-JmxTransSpec-reference[`JmxTransSpec` schema reference].

|xref:type-JmxTransOutputDefinitionTemplate-{context}[`JmxTransOutputDefinitionTemplate`] array

> |queries            1.2+<.<|Queries to send to the Kafka brokers to define what data should be read from each broker. For more information, see `proc-jmxtrans-deployment.adoc`.

Reviewer: Same comment as previous

Reviewer: See previous comment

|xref:type-JmxTransQueryTemplate-{context}[`JmxTransQueryTemplate`] array
|resources          1.2+<.<|CPU and memory resources to reserve.
|xref:type-ResourceRequirements-{context}[`ResourceRequirements`]
|====

[id='type-JmxTransOutputDefinitionTemplate-{context}']
### `JmxTransOutputDefinitionTemplate` schema reference

Used in: xref:type-JmxTransSpec-{context}[`JmxTransSpec`]

[options="header"]
|====
|Property           |Description
|outputType  1.2+<.<|Template for setting the format of the data that will be pushed.
|string

> |host        1.2+<.<|The ip of the remote host that the data will be pushed to.

Reviewer: ```suggestion
|host        1.2+<.<|The IP address of the remote host that the data is pushed to.
```

|string

> |port        1.2+<.<|The port of the remote host that the data will be pushed tp.

Reviewer 2: Typo: pushed to.

> |port        1.2+<.<|The port of the remote host that the data will be pushed to.

Reviewer: ```suggestion
|port        1.2+<.<|The port of the remote host that the data is pushed to.
```

|integer

> |flushDelay  1.2+<.<|How many seconds the JmxTransSpec will wait before push a new set of data out.

Reviewer 2: before push a new set of data out > before pushing a new set of data out.

> |flushDelay  1.2+<.<|How many seconds the JmxTransSpec will wait before pushing a new set of data out.

Reviewer: ```suggestion
|flushDelay  1.2+<.<|How many seconds the JmxTransSpec waits before pushing a new set of data out.
```

|integer

> |typeNames   1.2+<.<|Template for being able to filter out exactly what data you want when a wildcard query is sent. More information can be found here: https://github.com/jmxtrans/jmxtrans/wiki/Queries.

Reviewer: ```suggestion
|typeNames   1.2+<.<|Template for filtering data when a wildcard query is sent. For more information, go to https://github.com/jmxtrans/jmxtrans/wiki/Queries[JmxTrans queries].
```

|string array

> |name        1.2+<.<|Template for setting the name of the output definition which will be used to know where to send the data of the queries to.

Reviewer: ```suggestion
|name        1.2+<.<|Template for setting the name of the output definition, which is used to identify where to send the data of the queries to.
```

|string
|====

[id='type-JmxTransQueryTemplate-{context}']
### `JmxTransQueryTemplate` schema reference

Used in: xref:type-JmxTransSpec-{context}[`JmxTransSpec`]

[options="header"]
|====
|Property            |Description

> |targetMBean  1.2+<.<|If using a wildcard, the MBean(s) hat the data will be gathered from.

Reviewer: ```suggestion
|targetMBean  1.2+<.<|If using a wildcard, the MBean(s) hat the data is gathered from.
```

|string

> |attributes   1.2+<.<|attributes determine which attributes of the targeted MBean will be read.

Reviewer: ```suggestion
|attributes   1.2+<.<|Determine which attributes of the targeted MBean are read.
```

|string array

> |outputs      1.2+<.<|outputs is a list of names of the Output Definitions which configures where the data will be pushed to and what data format it will be in.

Reviewer: ```suggestion
|outputs      1.2+<.<|List of names of output definitions, which specify where JMX metrics are pushed to, and in which data format
```

|string array
|====

[id='type-ResourceRequirements-{context}']
### `ResourceRequirements` schema reference

Used in: xref:type-EntityTopicOperatorSpec-{context}[`EntityTopicOperatorSpec`], xref:type-EntityUserOperatorSpec-{context}[`EntityUserOperatorSpec`], xref:type-JmxTransSpec-{context}[`JmxTransSpec`], xref:type-KafkaBridgeSpec-{context}[`KafkaBridgeSpec`], xref:type-KafkaClusterSpec-{context}[`KafkaClusterSpec`], xref:type-KafkaConnectS2ISpec-{context}[`KafkaConnectS2ISpec`], xref:type-KafkaConnectSpec-{context}[`KafkaConnectSpec`], xref:type-KafkaExporterSpec-{context}[`KafkaExporterSpec`], xref:type-KafkaMirrorMakerSpec-{context}[`KafkaMirrorMakerSpec`], xref:type-TlsSidecar-{context}[`TlsSidecar`], xref:type-TopicOperatorSpec-{context}[`TopicOperatorSpec`], xref:type-ZookeeperClusterSpec-{context}[`ZookeeperClusterSpec`]

> [options="header"]

Reviewer: I think we're missing the descriptions here.

![image](https://user-images.githubusercontent.com/47596553/71826697-666f0180-3096-11ea-9b7f-2fde32d85ae5.png)

Reviewer: I'm not sure where the descriptions are in the code? This table is for `ResourceRequirements` schema reference

Author: There doesn't seem to be any description of those in master branch either? https://github.com/strimzi/strimzi-kafka-operator/blob/master/documentation/modules/appendix_crds.adoc

|string (one of [replace-key, renew-certificate])
|====

[id='type-JmxTransSpec-{context}']
### `JmxTransSpec` schema reference

Used in: xref:type-KafkaSpec-{context}[`KafkaSpec`]

[options="header"]
|====
|Property                  |Description
|image              1.2+<.<|The image to use for the JmxTrans.
|string
|outputDefinitions  1.2+<.<|Defines the output hosts that will be referenced later on. For more information on these properties see, xref:type-JmxTransSpec-reference[`JmxTransSpec` schema reference].
|xref:type-JmxTransOutputDefinitionTemplate-{context}[`JmxTransOutputDefinitionTemplate`] array

> |kafkaQueries       1.2+<.<|Queries to send to the Kafka brokers to define what data should be read from each broker. For more information on these properties see, xref:type-JmxTransSpec-reference[`JmxTransSpec` schema reference].

Reviewer 3: @PaulRMellor See here the circular linking. I don't think this makes sense. Is this really want you intended?
